application/SFTP.py

```diff
@@ -112,11 +112,13 @@ def rename(self, cwd, old, new):
 
         return True, ''
 
-    def remode(self, cwd, new_mode):
-        try:
-            self.sftp.chmod(cwd, new_mode)
-        except Exception as e:
-            return False, repr(e)
+    def chmod(self, path, mode, recursive):
+        _, _, _, stderr = self.exec_command_blocking(
+            f'chmod {"-R" if recursive else ""} {"{0:0{1}o}".format(mode,3)} "{path}"')
```

Comment:
My bad for suggesting the original approach of using Paramiko's APIs. 
The APIs don't support recusive operations natively. It is possible for us to list all the files in some dir and operate on each file, but the performance is very poor given each operation is an RPC and there are a lot of behaviours we need to mimic (e.g. should we proceed to the other files if we can't delete some file/dir in some dir). 
so I guess we better use the command line utilities instead?